backends/platform/sdl/sdl.cpp

```diff
@@ -657,6 +657,17 @@ Common::WriteStream *OSystem_SDL::createLogFile() {
 
 Common::String OSystem_SDL::getSystemLanguage() const {
 #if defined(USE_DETECTLANG) && !defined(WIN32)
+
+#if SDL_VERSION_ATLEAST(2, 0, 14)
+	SDL_Locale *locales = SDL_GetPreferredLocales();
+	if (locales) {
+		if (locales[0].language != NULL) {
+			return Common::String::format("%s_%s", locales[0].country, locales[0].language);
+		}
+		SDL_free(locales);
+	}
+#endif // SDL_VERSION_ATLEAST(2, 0, 14)
```

Comment:
This should go outside of the `#if defined(USE_DETECTLANG) && !defined(WIN32)` check so that it can be used on more platforms.

Reply:
with SDL 2.30.9 a lot of platform have it: https://github.com/libsdl-org/SDL/tree/release-2.30.9/src/locale
and PSP come on next version: https://github.com/libsdl-org/SDL/commit/dccd23e6fb6e3f98b61d2d605c5cd125611719b8

You can use it for macos: https://github.com/scummvm/scummvm/blob/master/backends/platform/sdl/macosx/macosx.cpp#L176 
and i think "win32": https://github.com/scummvm/scummvm/blob/master/backends/platform/sdl/win32/win32.cpp#L238